usl_pipeline/cloud_functions/main.py

```diff
@@ -313,15 +313,23 @@ def build_feature_matrix(cloud_event: functions_framework.CloudEvent) -> None:
 def _build_feature_matrix(
     bucket_name: str, chunk_name: str, output_bucket: str
 ) -> None:
-    """Builds a feature matrix when an archive of geo files is uploaded."""
+    """Builds a feature matrix when a set of geo files is uploaded."""
     storage_client = storage.Client()
     bucket = storage_client.bucket(bucket_name)
     chunk_blob = bucket.blob(chunk_name)
 
-    with chunk_blob.open("rb") as archive:
-        feature_matrix, metadata = _build_feature_matrix_from_archive(archive)
-        if feature_matrix is None:
-            raise ValueError(f"Empty archive found in {chunk_blob}")
+    with chunk_blob.open("rb") as chunk:
+        # Flood (CityCat)
+        if chunk_name.endswith(".tar"):
+            feature_matrix, metadata = _build_feature_matrix_from_archive(chunk)
+
+            if feature_matrix is None:
+                raise ValueError(f"Empty archive found in {chunk_blob}")
+        # Heat (WRF)
+        elif re.search(file_names.WPS_DOMAIN3_NC_REGEX, chunk_name):
```

Comment:
It'd also be nice to handle WPS files inside tars. If we can, I'd try to structure this as something like
```python
def _build_feature_matrix(bucket_name, chunk_name, output_bucket):
  if chunk_name.endswith(".tar"):
    tensors = []
    for member in get_tar_members(chunk):
      tensors.append(get_features(chunk))
    return numpy.dstack(tensors)
  else:
    return get_features(chunk)

def get_features(chunk):
  if name == file_names.ELEVATION_TIF:
    return = _read_elevation_features(chunk)
  elif name == file_names.BOUNDARIES_TXT:
    return _read_polygons_from_byte_stream(chunk)
```

Reply:
Ack - let me see if we can refactor this a bit more. I'll do it in a separate branch